appcelerator/titanium_mobile
# feat(android): replace Support libraries with AndroidX

**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-26472
- https://jira.appcelerator.org/browse/TIMOB-27696

**Summary:**
- Replaced all Google Support library references with AndroidX.
- Updated to newest version of all Google libraries.
- Enabled "jetifier" tool support for app builds.
  * Allows backward compatibility with old modules using Google Support libraries.
- [TIMOB-27696](https://jira.appcelerator.org/browse/TIMOB-27696) Added ability to extend app's "gradle.properties" file.
  * Can now add the following file to project: `./platform/android/gradle.properties`

---
**Test AndroidX:**
_(Perform test on Android device and emulator. Also test on Android 4.4.)_
1. Build and run [kitchensink-v2](https://github.com/appcelerator/kitchensink-v2) on Android.
2. Tap on every row, verifying that all features work.

---
**Test "gradle.properties" Extension:**
1. Create a Classic Titanium app project.
2. Add the below `gradle.properties` file under project's `./platform/android` dir.
3. Add the below `build.gradle` file under project's `./platform/android` dir.
4. Build for Android.
5. In the build log, look for message: `@@@ my.property = Hello World`

./platform/android/gradle.properties
```properties
my.property=Hello World
```

./platform/android/build.gradle
```groovy
println('@@@ my.property = ' + project.findProperty('my.property'))
```
